### PR TITLE
CHI-2421-case_section_crud

### PR DESCRIPTION
**Pull request by stephenhand**
## Description

* Adds CRUD endpoints for managing individual sections of cases
* Adds a PUT endpoint specifically for updating the case 'overview', i.e. the case 'info' blob
* Adds simplified permissions management for the above
* Marks the case PUT endpoint for deprecation - clients should be migrated to the above enpoints for all case editing operations
* Consolidated the separate permissions to edit case summary, follow up date and child is at risk flag into a single 'Edit Case Overview' permission

### Checklist
- [X] Corresponding issue has been opened
- [X] New tests added

### Related Issues
CHI-2421

### Verification steps

Automated service tests are included in the PR

### CODE REVIEW

File: hrm-domain/hrm-core/case/caseDataAccess.ts [hunk 1]
```diff
@@ -123,29 +125,24 @@ export type CaseListFilters = {
...
-          allSections,
-        )};${selectSingleCaseByIdSql('Cases')}`;
-        const queryValues = { accountSid, workerSid, caseId: inserted.id };
-        inserted = await transaction.one(sectionStatement, queryValues);
+      // eslint-disable-next-line @typescript-eslint/dot-notation
+      if ((caseRecord['caseSections'] ?? []).length) {
+        // No compatibility needed here as flex doesn't create cases with sections
+        console.warn(
+          `[DEPRECATION WARNING] Support for creating case sections with a case has been removed as of HRM v1.15.0. Add case sections using the dedicated case section CRUD endpoints going forward.`,
+        );
```
Comment (murilovmachado) [inline hrm-domain/hrm-core/case/caseDataAccess.ts:145]: Thanks for removing deprecated code

File: hrm-domain/hrm-core/case/caseSection/caseSectionRoutesV0.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    user.workerSid,
+  );
+
+  res.json(createdCase);
+});
+
+caseSectionsRouter.put(
+  '/:sectionType/:sectionId',
+  canEditCaseSection,
+  async (req, res) => {
```
Comment (murilovmachado) [inline hrm-domain/hrm-core/case/caseSection/caseSectionRoutesV0.ts:84]: Very nice to define the case sections' endpoints like this, with `:sectionId`, so we don't have to manually create a bunch of similar endpoints.

**Review by murilovmachado — APPROVED** [review]
Looks great!! 🚀

File: hrm-domain/hrm-core/case/caseRoutesV0.ts [hunk 3]
```diff
@@ -102,6 +112,44 @@ casesRouter.put('/:id/status', canEditCase, async (req, res) => {
...
+  const { id } = req.params;
+  const updatedCase = await caseApi.updateCaseStatus(id, status, accountSid, {
+    can: req.can,
+    user,
+  });
+  if (!updatedCase) {
+    throw createError(404);
+  }
+  res.json(updatedCase);
+});
```
Comment (murilovmachado) [inline hrm-domain/hrm-core/case/caseRoutesV0.ts:130]: Is this endpoint duplicated?

Comment (stephenhand) [reply hrm-domain/hrm-core/case/caseRoutesV0.ts:130]: Looks like, not sure what happened there, nice catch!

**Pull request merged**